src/Symfony/Bridge/PhpUnit/SymfonyTestsListener.php

```diff
@@ -197,6 +197,11 @@ public function endTest(\PHPUnit_Framework_Test $test, $time)
                 DnsMock::withMockedHosts(array());
             }
         }
+
+        if ($test instanceof \PHPUnit_Framework_TestCase && 0 === strpos($test->getName(), 'testLegacy') && empty($test->getTestResultObject()->warningCount())) {
```

Comment:
Adding the warning based on class/method names does work. I am looking into how to do that (if possible) based on the data transformer.